engine/execution/state/state.go

```diff
@@ -363,6 +363,9 @@ func (s *state) GetExecutionResultID(ctx context.Context, blockID flow.Identifie
 	return result.ID(), nil
 }
 
+// SaveExecutionResults saves all data related to the execution of a block.
+// It is concurrent-safe
```

Comment:
It is concurrent safe because chunk data packs store is conflict-free (storing data by hash), and protocol data requires a lock to store, which will be synchronized.